pkg/rehearse/jobs.go

```diff
@@ -364,7 +376,9 @@ func (jc *JobConfigurer) ConfigurePeriodicRehearsals(periodics config.Periodics)
 			metadata.Repo = job.ExtraRefs[0].Repo
 			metadata.Branch = job.ExtraRefs[0].BaseRef
 		}
-		if err := jc.configureJobSpec(job.Spec, metadata, jc.loggers.Debug.WithField("name", job.Name)); err != nil {
+		jobNameWithoutTest := metadata.JobName(jobconfig.PeriodicPrefix, "")
```

Comment:
Can we add a method to `Metadata` to get the test name instead? This is part of the API.

Reply:
I think we already have that:

https://github.com/openshift/ci-tools/blob/307b5c731889f13777fa0e361cdb05418ab72237/pkg/api/metadata.go#L39-L46

Reply:
Not quite ... we have a "test name" meaning the identifier devs use like "unit", then a "test name" as used in `/test foo` like "variant-unit" and a "test name" as in `pull-ci-org-repo-branch-variant-name` ... that function you linked gives you the second from the first, I want the first from the last.

Reply:
The `TestName` method isn't useful for us in the rehearse code. That's just a helper function or `JobName` that takes the `testName` and appends a `-variant` if the metadata contains a variant or just the given `testName` unmodified if there is no variant.

This does make me realize that the code I wrote doesn't account for variants though. I'll make a function in `Metadata` that can handle all of this including variants.